monitoring/rid_qualifier/flight_track_factory.py

```diff
@@ -0,0 +1,165 @@
+
+        ''' This method checks if the input extents are valid i.e. small enough, if the extent is too large, we reject them, at the moment it checks for extents less than 500m x 500m square but can be changed as necessary.''' 
+
+        box = shapely.geometry.box(self.minx, self.miny, self.maxx, self.maxy)
+        geod = Geod(ellps="WGS84")
+        area = abs(geod.geometry_area_perimeter(box)[0])
+        if (area) < 250000: # Have a area less than 500m x 500m square
+            return True
+        else: 
+            return False
+        
+    def generate_query_bboxes(self):
+        ''' For the differnet Remote ID checks: No, we need to generate three bounding boxes for the display provider, this method generates the 1 km diagonal length bounding box '''
+        # Get center of of the bounding box that is inputted into the generator
+        box = shapely.geometry.box(self.minx, self.miny, self.maxx, self.maxy)
+        center = box.centroid
+        # Transform to geographic co-ordinates to get areas
+        transformer = Transformer.from_crs("epsg:4326", "epsg:3857")
+        transformed_x,transformed_y = transformer.transform(center.x, center.y)
+        pt = Point(transformed_x,transformed_y)
+        # Now we have a point, we can buffer the point and create bounding boxes of the buffer to get the appropriate polygons, more than three boxes can be created, for the tests three will suffice.
+        box_diagonals = {1:440,2:900,3:3000}
+
+        for box_id, box_diagonal in box_diagonals.items():
+            # Buffer the point with the appropriate length
+            buffer = pt.buffer(box_diagonal)                   
+            buffer_bounds = buffer.bounds
+            buffer_bounds_polygon  =  shapely.geometry.box(buffer_bounds[0], buffer_bounds[1], buffer_bounds[2], buffer_bounds[3])
+            buffer_points =  zip(buffer_bounds_polygon.exterior.coords.xy[0], buffer_bounds_polygon.exterior.coords.xy[1])
+            proj_buffer_points = []
+            # reproject back to ESPG 4326
+            transformer2 = Transformer.from_crs("epsg:3857", "epsg:4326")
+            for point in buffer_points:
+                x = point[0]
+                y = point[1]
+                x, y = transformer2.transform(x, y)
+                proj_buffer_points.append((x, y))
+            
+            buffered_box = Polygon(proj_buffer_points)
+            # Get the bounds of the buffered box, this is the one that will will be fed to the remote ID display provider to query
+            buffered_box_bounds = buffered_box.bounds
+            self.query_bboxes.append({'bounds':buffered_box_bounds,'shape':buffered_box})
+            
+
+    def generate_flight_grid(self):
+        ''' Generate a series of boxes within the bounding box to have areas for different flights '''
+
+        cell_size = (self.maxx-self.minx)/(self.max_cols//2)        
+        grid_cells = []
+        for x0 in np.arange(self.minx, self.maxx+cell_size, cell_size ):
```

Comment:
I have refactored the code so that Numpy dependency is removed and we build a 3x2 grid to begin with (total 6 flights). We can make it customizable later.